tests/unit_tests/CMakeLists.txt

```diff
@@ -59,6 +59,9 @@ foreach(domain ${TARGET_DOMAINS})
   add_executable(test_main_${domain}_ct main_test.cpp)
   target_include_directories(test_main_${domain}_ct PUBLIC ${GTEST_INCLUDE_DIR})
   target_compile_options(test_main_${domain}_ct PRIVATE -fsycl)
+  if (USE_ADD_SYCL_TO_TARGET_INTEGRATION)
+    add_sycl_to_target(TARGET test_main_${domain}_ct SOURCES t main_test.cpp)
```

Comment:
`t` seems like a good source file name..? :D

also, shouldn't the setting of the `-fsycl` flag be moved to the else branch (potentially even only if the implementation option is set to `dpc++`)?